src/main/java/de/tum/in/www1/artemis/service/UserService.java

```diff
@@ -418,8 +418,18 @@ public String decryptPassword() {
      * @param login of a user
      * @return decrypted password or empty string
      */
-    public Optional<String> decryptPasswordByLogin(String login) {
-        return userRepository.findOneByLogin(login).map(u -> encryptor().decrypt(u.getPassword()));
+    public Optional<char[]> decryptPasswordByLogin(String login) {
+        return userRepository.findOneByLogin(login).map(this::decryptPasswordOfUser);
+    }
+
+    /**
+     * Returns the password of the user as a char array (as is the Java convention for handling sensitive Strings)
+     *
+     * @param user The user for whom to fetch the password
+     * @return The password of the specified user
+     */
+    public char[] decryptPasswordOfUser(User user) {
```

Comment:
why do we need a second method here? I would prefer if we only have one method and if it uses a String return type

Reply:
From my other response:
> I kept this method and renamed it, because the decryptPasswordByLogin method also fetches the user from the database, while we already have the user in this case.
In theory, the user is probably cached if we just call the other method, so it's not that expensive. Do you want me to remove this?

Reply:
Yes, I still prefer to have only one method. This is a critical method and I would like to reduce the maintenance effort for such critical code.

`public Optional<String> decryptPasswordByLogin(String login)`

If the user needs to be fetched, we can do this in a separate call and then invoke this method with `user.getLogin()`.

Reply:
I reverted the changes